sycl/source/builtins/relational_functions.cpp

```diff
@@ -16,6 +16,15 @@
 namespace sycl {
 inline namespace _V1 {
 
+template <typename T> static auto process_arg_for_macos(T x) {
```

Comment:
should this function be declared constexpr?  I think it's not strictly required, but if that's its sole intended use, might be best.

Reply:
`x` is a run-time value most of the time...